### PR TITLE
API:Watchlist feed code samples, plural

### PR DESCRIPTION
**Pull request by martyav**

### CODE REVIEW

File: python/feedwatchlist_self.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+PARAMS_2 = {
+    "action": "login",
+    "lgname": "user_name",
+    "lgpassword": "password",
+    "format": "json",
+    "lgtoken": LOGIN_TOKEN
+}
+
+R = S.post(URL, data=PARAMS_2)
+DATA = R.json()
```
Comment (srish) [inline python/feedwatchlist_self.py]: We don't need this line here.

File: python/feedwatchlist_self.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+R = S.get(url=URL, params=PARAMS_1)
+DATA = R.json()
+
+LOGIN_TOKEN = DATA['query']['tokens']['logintoken']
+
+# Step 2: Send a post request to log in. For this login 
+# method, Obtain credentials by first visiting
+# https://www.test.wikipedia.org/wiki/Manual:Bot_passwords
```
Comment (srish) [inline python/feedwatchlist_self.py]: The URL here is inconsistent with API endpoint you are using above.

File: python/feedwatchlist_other.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+URL = "https://en.wikipedia.org/w/api.php"
+
+PARAMS = {
+    "action": "feedwatchlist",
+    "wlowner": "sample_user",
+    "wltoken": "sample_watchlist_token"
+}
+
+R = S.get(url=URL, params=PARAMS)
+DATA = R.json()
```
Comment (srish) [inline python/feedwatchlist_other.py]: I tested this code, and it looks like we won't be able to generate other user's watchlist with our watchlist token. This is not very clear and super confusing in the original docs, and we should address this in the doc in progress what are the alternatives to this. The user experience around obtaining a watchlist token on MediaWiki.org (https://www.mediawiki.org/wiki/Special:Preferences#mw-prefsection-watchlist) is confusing too; I think we should create a task about this on Phabricator.

One TODO here would also be to change `R.json()`
 to support fetching the output which will be in XML format and not json, unlike the other API modules.

File: python/feedwatchlist_self.py (hunk 1 above)
Comment (martyav) [reply python/feedwatchlist_self.py]: Ok. This has been removed from the renamed file, get_my_watchlist_feed.py

File: python/feedwatchlist_self.py (hunk 2 above)
Comment (martyav) [reply python/feedwatchlist_self.py]: Updated in renamed file, get_my_watchlist_feed.py

File: python/feedwatchlist_other.py (hunk 3 above)
Comment (martyav) [reply python/feedwatchlist_other.py]: Woops! I swear an earlier version had R.text there. Updated in renamed file, get_other_watchlist_feed.py

File: python/get_user_watchlist_feed.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#This file is partly auto-generated
+
+#!/usr/bin/python3
+
+"""
+    feedwatchlist_other.py
```
Comment (srish) [inline python/get_user_watchlist_feed.py]: Filename in the code is different than the actual.

File: python/get_my_watchlist_feed.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python3
+
+"""
+    get_my_watchlist.py
```
Comment (srish) [inline python/get_my_watchlist_feed.py]: File name differs here..

File: modules.json [hunk 6]
```diff
@@ -332,5 +332,15 @@
             "format": "json",
             "modules": "parse|query+info|query+blah"
         }
+    },
+    {
+        "filename": "feedwatchlist_other.py",
```
Comment (srish) [inline modules.json]: File name is different..

File: python/get_user_watchlist_feed.py (hunk 4 above)
Comment (martyav) [reply python/get_user_watchlist_feed.py]: Fixed in https://github.com/srish/MediaWiki-Action-API-Code-Samples/pull/42/commits/6d0834180dc5c959cb016f363a1c0eb046357ba5

File: python/get_my_watchlist_feed.py (hunk 5 above)
Comment (martyav) [reply python/get_my_watchlist_feed.py]: Fixed in https://github.com/srish/MediaWiki-Action-API-Code-Samples/pull/42/commits/6d0834180dc5c959cb016f363a1c0eb046357ba5

File: modules.json (hunk 6 above)
Comment (martyav) [reply modules.json]: Fixed in https://github.com/srish/MediaWiki-Action-API-Code-Samples/pull/42/commits/6d0834180dc5c959cb016f363a1c0eb046357ba5

**Pull request merged**